cim_schema_import.go

```diff
@@ -0,0 +1,583 @@
+				cimTypes[cimType.Id] = &cimType
+			}
+		} else if strings.Contains(objType, "http://www.w3.org/1999/02/22-rdf-syntax-ns#Property") {
+			cimAttribute := processProperty(v)
+			cimAttribute.Origin = cimOntology.Keyword
+			cimAttribute.Origins = []string{cimOntology.Keyword}
+			cimAttributes = append(cimAttributes, &cimAttribute)
+		} else if strings.Contains(objType, "http://iec.ch/TC57/1999/rdf-schema-extensions-19990926#ClassCategory") {
+
+		} else if strings.Contains(objType, "http://www.w3.org/2002/07/owl#Ontology") {
+			cimOntology = processOntology(v)
+		} else {
+			cimEnumValue := processEnumValue(v)
+			cimEnumValues = append(cimEnumValues, &cimEnumValue)
+		}
+	}
+
+	for _, attr := range cimAttributes {
+		if v, ok := cimTypes[attr.RDFDomain]; ok {
+			attr.Categories = v.Categories
+			v.Attributes = append(v.Attributes, attr)
+		}
+	}
+
+	for _, val := range cimEnumValues {
+		if enum, ok := cimEnums[val.RDFType]; ok {
+			enum.Values = append(enum.Values, val)
+		}
+	}
+
+	return cimTypes, cimEnums, cimOntology
+}
+
+func mergeCimTypes(typesMerged map[string]*CIMType, types map[string]*CIMType) map[string]*CIMType {
+	for k := range types {
+		if v, ok := typesMerged[k]; ok {
+			if types[k].SuperType != "" {
+				v.SuperType = types[k].SuperType
+			}
+
+			if types[k].Stereotype != "" {
+				v.Stereotype = types[k].Stereotype
+			}
+
+			if len(types[k].Categories) > 0 {
+				v.Categories = append(v.Categories, types[k].Categories...)
+			}
+
+			if types[k].Origin != "" {
+				v.Origins = append(v.Origins, types[k].Origin)
+			}
+
+			for _, attr := range types[k].Attributes {
+				if existingAttrIndex := FindCIMAttributeById(v.Attributes, attr.Id); existingAttrIndex != -1 {
+					// If the attribute already exists, we can merge the attributes
+					existingAttr := v.Attributes[existingAttrIndex]
+					existingAttr.Origins = append(existingAttr.Origins, attr.Origin)
+					existingAttr.Categories = append(existingAttr.Categories, attr.Categories...)
+				} else {
+					v.Attributes = append(v.Attributes, attr)
+				}
+			}
+		} else {
+			typesMerged[k] = types[k]
+		}
+	}
+	return typesMerged
+}
+
+// FindCIMAttributeById searches for a CIMAttribute with the given Id in a slice.
+func FindCIMAttributeById(attrs []*CIMAttribute, id string) int {
+	for i, attr := range attrs {
+		if attr.Id == id {
+			return i
+		}
+	}
+	return -1
+}
+
+func mergeCimEnums(typesMerged map[string]*CIMEnum, types map[string]*CIMEnum) map[string]*CIMEnum {
+	for k := range types {
+		if v, ok := typesMerged[k]; ok {
+			if types[k].Stereotype != "" {
+				v.Stereotype = types[k].Stereotype
+			}
+
+			if types[k].Origin != "" {
+				v.Origins = append(v.Origins, types[k].Origin)
+			}
+
+			for _, val := range types[k].Values {
+				if existingValIndex := FindCIMEnumValueById(v.Values, val.Id); existingValIndex == -1 {
+					v.Values = append(v.Values, val)
+				}
+			}
+		} else {
+			typesMerged[k] = types[k]
+		}
+	}
+	return typesMerged
+}
```

Comment:
Rename `typesMerged` to `enumsMerged` and `types` to `enums`.
```suggestion
func mergeCimEnums(enumsMerged map[string]*CIMEnum, enums map[string]*CIMEnum) map[string]*CIMEnum {
	for k := range enums {
		if v, ok := enumsMerged[k]; ok {
			if enums[k].Stereotype != "" {
				v.Stereotype = enums[k].Stereotype
			}

			if enums[k].Origin != "" {
				v.Origins = append(v.Origins, enums[k].Origin)
			}

			for _, val := range enums[k].Values {
				if existingValIndex := FindCIMEnumValueById(v.Values, val.Id); existingValIndex == -1 {
					v.Values = append(v.Values, val)
				}
			}
		} else {
			enumsMerged[k] = enums[k]
		}
	}
	return enumsMerged
}
```